pkg/codegen/nodejs/gen.go

```diff
@@ -1125,12 +1139,21 @@ func (mod *modContext) genIndex(exports []string) string {
 		}
 	}
 
-	var children []string
+	children := codegen.NewStringSet()
+	versionSuffix := regexp.MustCompile(`/v\d+((alpha|beta)\d+)?`)
 	for _, mod := range mod.children {
-		children = append(children, mod.mod)
+		child := mod.mod
+		// Trim version suffix from child modules. Nested versions will have their own index.ts file.
+		if versionSuffix.MatchString(child) {
+			if i := strings.LastIndex(child, "/"); i > 0 {
+				child = child[:i]
+			}
+		}
+		children.Add(child)
 	}
```

Comment:
This seems pretty k8s-specific. What's the concrete effect of this code?

Reply:
It changes the import paths. 
```diff
-import * as apps from "./apps";
-export {apps};
+import * as apps/v1 from "./apps/v1";
+import * as apps/v1beta1 from "./apps/v1beta1";
+import * as apps/v1beta2 from "./apps/v1beta2";
+export {apps/v1, apps/v1beta1, apps/v1beta2};
```

Reply:
Got it. Can we put it under the k8s compatibility mode?